src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,249 @@
+        listAdapter = new WifiListAdapter();
+        mListWifi.setAdapter(listAdapter);
+    }
+
+    /**
+     * Load the configured WiFi networks, sort them by SSID.
+     *
+     * @return a sorted array of WifiConfiguration, or null, if data cannot be retrieved
+     */
+    private WifiConfiguration[] loadConfiguredNetworksSorted() {
+        WifiManager wifiManager = (WifiManager) getContext().getSystemService(Context.WIFI_SERVICE);
+        if (wifiManager != null) {
+            List<WifiConfiguration> configuredNetworks = wifiManager.getConfiguredNetworks();
+            // if WiFi is turned off, getConfiguredNetworks returns null on many devices
+            if (configuredNetworks != null) {
+                WifiConfiguration[] result = configuredNetworks.toArray(new WifiConfiguration[configuredNetworks.size()]);
+                Arrays.sort(result, new Comparator<WifiConfiguration>() {
+                    @Override
+                    public int compare(WifiConfiguration lhs, WifiConfiguration rhs) {
+                        return lhs.SSID.compareToIgnoreCase(rhs.SSID);
+                    }
+                });
+                return result;
+            }
+        }
+        // WiFi is turned of or device doesn't have WiFi
+        return null;
+    }
+
+    @Override
+    protected void showDialog(Bundle state) {
+        super.showDialog(state);
+        updateSaveButtonState();
+    }
+
+    private void updateSaveButtonState() {
+        Dialog dialog = getDialog();
+        if (dialog instanceof AlertDialog) {
+            AlertDialog alert = (AlertDialog) dialog;
+            // enabled if the WiFi information is available
+            boolean saveEnabled = mConfiguredNetworks != null;
+            alert.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(saveEnabled);
+        }
+    }
+
+    /**
+     * Responsible for persisting (if positiveResult).
+     */
+    @Override
+    protected void onDialogClosed(boolean positiveResult) {
+        super.onDialogClosed(positiveResult);
+
+        if (positiveResult && isPersistent()) {
+            // Defensive copy of SSID names, as set is then owned by the SharedPreference
+            // getEditor() returns new object every time, so need to store
+            Set<String> ssids = new HashSet<>(mSelectedSsids);
+            invokeOnPreferenceChange(ssids);
+            SharedPreferences.Editor editor = getEditor();
+            editor.putStringSet(getKey(), ssids);
+            if (shouldCommit()) {
+                editor.commit();
+            }
+        }
+    }
+
+    private void invokeOnPreferenceChange(Set<String> ssids) {
+        OnPreferenceChangeListener listener = getOnPreferenceChangeListener();
+        if (listener != null) {
+            listener.onPreferenceChange(this, ssids);
+        }
+    }
+
+    /**
+     * Adapter between the available and selected Wifis in the preference.
+     */
+    class WifiListAdapter extends BaseAdapter {
```

Comment:
You should use an ArrayAdapter, and pass the array in instead of accessing it directly. Means you only have to implement getView(), nothing else.